src/WebJobs.Script.WebHost/Filters/ArmExtensionResourceFilter.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Controllers;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Azure.WebJobs.Script.Extensions;
+using Microsoft.Azure.WebJobs.Script.WebHost.Properties;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Filters
+{
+    /// <summary>
+    /// Resource filter used to ensure secrets aren't returned for GET requests made via the Functions ARM extension
+    /// API (hostruntime), unless properly authorized.
+    /// </summary>
+    public sealed class ArmExtensionResourceFilter : IAsyncResourceFilter
+    {
+        public async Task OnResourceExecutionAsync(ResourceExecutingContext context, ResourceExecutionDelegate next)
+        {
+            bool isArmExtensionRequest = !string.IsNullOrEmpty(context.HttpContext.Request.GetHeaderValueOrDefault(ScriptConstants.AntaresARMExtensionsRouteHeader));
+            bool isOwnerOrCoAdmin = string.Equals(context.HttpContext.Request.GetHeaderValueOrDefault(ScriptConstants.AntaresClientAuthorizationSourceHeader), "legacy", StringComparison.OrdinalIgnoreCase);
+            bool isRoleBasedContributor = string.Equals(context.HttpContext.Request.GetHeaderValueOrDefault(ScriptConstants.AntaresRoleBasedContributorHeader), "1");
+
+            if (isArmExtensionRequest && string.Equals(context.HttpContext.Request.Method, "GET", StringComparison.OrdinalIgnoreCase) &&
+                !isOwnerOrCoAdmin && !isRoleBasedContributor)
```

Comment:
These are the same Owner/RBAC checks that are already performed in Kudu for similar purposes [here](https://github.com/projectkudu/kudu/blob/e57833fea88995d6a24c8c06f956cc291838bdc1/Kudu.Services.Web/Tracing/TraceModule.cs#L74)